ads/google/imaVideo.js

```diff
@@ -0,0 +1,957 @@
+  fullscreenDiv.style.textAlign = 'center';
+  fullscreenDiv.style.fontWeight = 'bold';
+  fullscreenDiv.style.lineHeight = '1.4em';
+  fullscreenDiv.appendChild(global.document.createTextNode(fullscreenChars));
+  controlsDiv.appendChild(fullscreenDiv);
+
+  // Ad container.
+  adContainerDiv = global.document.createElement('div');
+  adContainerDiv.id = 'ima-ad-container';
+  adContainerDiv.style.position = 'absolute';
+  adContainerDiv.style.top = '0px';
+  adContainerDiv.style.left = '0px';
+  adContainerDiv.style.width = '100%';
+  adContainerDiv.style.height = '100%';
+
+  // Wraps our content video.
+  contentDiv = global.document.createElement('div');
+  contentDiv.id = 'ima-content';
+  contentDiv.style.position = 'absolute';
+  contentDiv.style.top = '0px';
+  contentDiv.style.left = '0px';
+  contentDiv.style.width = '100%';
+  contentDiv.style.height = '100%';
+  // The video player
+  videoPlayer = global.document.createElement('video');
+  videoPlayer.id = 'ima-content-player';
+  videoPlayer.style.width = '100%';
+  videoPlayer.style.height = '100%';
+  videoPlayer.style.backgroundColor = 'black';
+  videoPlayer.setAttribute('src', data.src);
+  videoPlayer.setAttribute('poster', data.poster);
+  videoPlayer.setAttribute('playsinline', true);
+
+  contentDiv.appendChild(videoPlayer);
+  wrapperDiv.appendChild(contentDiv);
+  wrapperDiv.appendChild(adContainerDiv);
+  wrapperDiv.appendChild(controlsDiv);
+  wrapperDiv.appendChild(bigPlayDiv);
+  global.document.getElementById('c').appendChild(wrapperDiv);
+
+  window.addEventListener('message', onMessage);
+
+  /**
+   * Set-up code that can't run until the IMA lib loads.
+   */
+  getIma(global, function() {
+    // This is the first place where we have access to any IMA objects.
+    contentComplete = false;
+    adsActive = false;
+    playbackStarted = false;
+    nativeFullscreen = false;
+
+    interactEvent = 'click';
+    mouseDownEvent = 'mousedown';
+    mouseMoveEvent = 'mousemove';
+    mouseUpEvent = 'mouseup';
+    if (navigator.userAgent.match(/iPhone/i) ||
+        navigator.userAgent.match(/iPad/i) ||
+        navigator.userAgent.match(/Android/i)) {
+      interactEvent = 'touchend';
+      mouseDownEvent = 'touchstart';
+      mouseMoveEvent = 'touchmove';
+      mouseUpEvent = 'touchend';
+    }
+    bigPlayDiv.addEventListener(interactEvent, onClick);
+    playPauseDiv.addEventListener(interactEvent, onPlayPauseClick);
+    progressBarWrapperDiv.addEventListener(mouseDownEvent, onProgressClick);
+    fullscreenDiv.addEventListener(interactEvent, onFullscreenClick);
+
+    const fullScreenEvents = [
+      'fullscreenchange',
+      'mozfullscreenchange',
+      'webkitfullscreenchange'];
+    for (const key in fullScreenEvents) {
+      global.document.addEventListener(
+          fullScreenEvents[key],
+          onFullscreenChange,
+          false);
+    }
+
+    adDisplayContainer = new google.ima.AdDisplayContainer(adContainerDiv,
```

Comment:
Seeking advice - the linter is complaining that google is not defined, and rightfully so, but it is defined by the ima3 library I load in getIma. How can I tell the linter that these will be defined at runtime by external code?